test/framework/resources/aws/utils/nodegroup.go

```diff
@@ -229,6 +229,24 @@ func GetClusterVPCConfig(f *framework.Framework) (*ClusterVPCConfig, error) {
 		PrivateSubnetList: []string{},
 	}
 
+	if len(f.Options.PublicSubnets) > 0 {
+		clusterConfig.PublicSubnetList = strings.Split(f.Options.PublicSubnets, ",")
+	}
+	if len(f.Options.PrivateSubnets) > 0 {
+		clusterConfig.PrivateSubnetList = strings.Split(f.Options.PublicSubnets, ",")
+	}
+	if len(f.Options.AvailabilityZones) > 0 {
+		clusterConfig.AvailZones = strings.Split(f.Options.AvailabilityZones, ",")
+	}
+	if f.Options.PublicRouteTableID != "" {
+		clusterConfig.PublicRouteTableID = f.Options.PublicRouteTableID
```

Comment:
Previously we used to get the `PublicRouteTableID` but with this logic user might end up just giving public subnet and we return in line 247. Should we maybe error out? Am I missing anything?

Reply:
Yeah,  I'll rework it and only bail if we have at least route table ID, public subnets and AZs